src/main/java/function/TaskList.java

```diff
@@ -0,0 +1,245 @@
+        if (taskListNotEmpty(taskArrayList)) {
+            printTaskListHeader();
+            printTaskListBody();
+        } else {
+            printErrorMessage(ERROR_NO_TASK);
+        }
+    }
+
+    /**
+     * Checks if taskArrayList is empty
+     *
+     * @param taskArrayList ArrayList of user's tasks
+     * @return boolean of whether it is not empty
+     */
+    public boolean taskListNotEmpty (ArrayList<Task> taskArrayList) {
+        return !(taskArrayList.isEmpty());
+    }
+
+    /**
+     * Prints the content of the task list with respective list index, task symbol, done symbol
+     * and description.
+     */
+    public void printTaskListBody() {
+        for (int listIndex = 1; listIndex <= taskArrayList.size(); listIndex++) {
+            Task currentTask = taskArrayList.get(listIndex - 1);
+            System.out.print(listIndex + ". ");
+            System.out.println(currentTask.toString());
+        }
+    }
+
+    /**
+     * Prints header of the task list with total number of tasks and number of completed tasks.
+     */
+    public void printTaskListHeader() {
+        System.out.println("Here is your current task list!");
+        System.out.println("You have " + getTotalNumberOfTasks() + " task"
+                + ((getTotalNumberOfTasks()>1)? "s" :"") + " on your list!");
+        System.out.println("You have completed " + getNumberOfCompleteTasks() +" of them.");
+        System.out.println("Hope you are on target!");
+    }
+
+    /**
+     * Takes a string of userInput and determines the TaskType intended.
+     * Creates an instance of the new TaskType,
+     * then, add the instance of the new task into taskArrayList.
+     *
+     * @param userInput name of new task.
+     */
+    public void addTask(String userInput) {
+        TaskType newTaskType = getInputTaskType(userInput);
+        Task newEntry;
+
+        if (taskTypeIsValid(newTaskType)) {
+            switch (newTaskType) {
+            case EVENT:
+                String eventName = getInputTaskName(userInput);
+                String period = getInputDetails(userInput);
+                newEntry = new Event(eventName, period);
+                break;
+            case DEADLINE:
+                String deadlineName = getInputTaskName(userInput);
+                String deadlineBy = getInputDetails(userInput);
+                newEntry = new Deadline(deadlineName, deadlineBy);
+                break;
+            case TODO:
+                int nameStartPoint = userInput.indexOf(" ");
+                String toDoName = userInput.substring(nameStartPoint);
+                newEntry = new ToDo(toDoName);
+                break;
+            default:
+                printErrorMessage(ERROR_ADDING_TASK);
+                return;
+            }
+            taskArrayList.add(newEntry);
+            printAddTaskSuccessfully(newEntry);
+        } else {
+                printErrorMessage(ERROR_ADDING_TASK);
+        }
```

Comment:
I think this else can be combine into the switch. Because the default operates the same thing as else, maybe do not need to check valid?